cocos/game/splash-screen.ts

```diff
@@ -307,12 +316,31 @@ export class SplashScreen {
         let scaleY = 1;
         // update bg uniform
         if (this.settings.background!.type === 'custom') {
-            if (dw < dh) {
-                scaleX = dh * this.bgRatio;
+            if (this.settings.policy === ResolutionPolicy.FIXED_WIDTH) {
+                scaleX = dw;
+                scaleY = (dw / this.bgImage.width) * this.bgImage.height;
+            } else if (this.settings.policy === ResolutionPolicy.FIXED_HEIGHT) {
+                scaleX = (dh / this.bgImage.height) * this.bgImage.width;
                 scaleY = dh;
+            } else if (this.settings.policy === ResolutionPolicy.SHOW_ALL) {
+                if (dw > dh) {
+                    scaleX = (dh / this.bgImage.height) * this.bgImage.width;
+                    scaleY = dh;
+                } else {
+                    scaleX = dw;
+                    scaleY = (dw / this.bgImage.width) * this.bgImage.height;
+                }
```

Comment:
The logic seems not correct, should choose the one the has small scale.

Reply:
Fixed